biz.aQute.bndlib/src/aQute/bnd/build/WorkspaceLock.java

```diff
@@ -103,6 +104,30 @@ private TimeoutException timeout(Lock lock) {
 		return e;
 	}
 
+	<T, U> T writeReadLocked(final long timeoutInMs, final Callable<U> underWrite,
+		final FunctionWithException<U, T> underRead, final BooleanSupplier canceled) throws Exception {
+		Callable<U> writeLocked = () -> {
+			U writeResult = underWrite.call();
+			// It is safe to let an exception propagate from here.
+			// Downgrade lock by taking read lock before returning to release
+			// write lock.
+			trace("Downgrading", writeLock());
```

Comment:
not the right comment is it? It is taking a read lock

Reply:
We are downgrading the write lock which is the current lock being held to a read lock. So the comment is correct.